### PR TITLE
[18.0][MIG] partner_company_group, base_partner_company_group, account_partner_company_group, crm_partner_company_group, sale_partner_company_group: Migration to 18.0

### PR DESCRIPTION
**Pull request by nguyenminhchien**
Some commits cannot be ported, i created a new commit for each module to add them to the blacklist:
- crm_partner_company_group: https://github.com/nguyenminhchien/partner-contact/commit/6c396679c87877e087803970363efcdadd9fe75d
- sale_partner_company_group: https://github.com/nguyenminhchien/partner-contact/commit/eaa6cd84023176cd52137e9a0c4296e26537b720
- partner_company_group: https://github.com/nguyenminhchien/partner-contact/commit/21226ac2d0b19b2f6b0cef6063e679eda7290c21

### CODE REVIEW

File: account_partner_company_group/readme/CREDITS.md [hunk 1]
```diff
@@ -0,0 +1 @@
+The migration of this module from 17.0 to 18.0 was financially supported by Camptocamp.
```
Comment (thienvh332) [inline account_partner_company_group/readme/CREDITS.md:1]: I don't see you add your name to contributors.txt. Your contribution is worth for future generations to admire. :smile:

File: crm_partner_company_group/readme/CREDITS.md [hunk 2]
```diff
@@ -0,0 +1 @@
+The migration of this module from 17.0 to 18.0 was financially supported by Camptocamp.
```
Comment (thienvh332) [inline crm_partner_company_group/readme/CREDITS.md:1]: Same as previous comment

File: sale_partner_company_group/readme/CREDITS.md [hunk 3]
```diff
@@ -0,0 +1 @@
+The migration of this module from 17.0 to 18.0 was financially supported by Camptocamp.
```
Comment (thienvh332) [inline sale_partner_company_group/readme/CREDITS.md:1]: Same as previous comment too

File: partner_company_group/readme/CREDITS.md [hunk 4]
```diff
@@ -0,0 +1 @@
+The migration of this module from 17.0 to 18.0 was financially supported by Camptocamp.
```
Comment (thienvh332) [inline partner_company_group/readme/CREDITS.md:1]: same as previous comment too too :laughing:

File: sale_partner_company_group/models/res_partner.py [hunk 5]
```diff
@@ -0,0 +1,60 @@
...
+            and self.company_group_id.property_product_pricelist
+            != self.property_product_pricelist
+        ):
+            price_list = self.company_group_id.property_product_pricelist
+            res["warning"] = {
+                "title": _("Warning"),
+                "message": _(
+                    "The company group %(company_group)s has"
+                    " the pricelist %(pricelist)s, that is different"
+                    " than the pricelist set on this contact"
```
Comment (thienvh332) [inline sale_partner_company_group/models/res_partner.py]:
```suggestion
                "title": self.env._("Warning"),
                "message": self.env._(
                    "The company group %(company_group)s has"
                    " the pricelist %(pricelist)s, that is different"
                    " than the pricelist set on this contact"
```

File: sale_partner_company_group/models/res_partner.py [hunk 6]
```diff
@@ -0,0 +1,60 @@
...
+            )
+            members_str = ""
+            for member in company_members.sorted(key="display_name"):
+                members_str += f"\t- {member.display_name}\n"
+            res["warning"] = {
+                "title": _("Warning"),
+                "message": _(
+                    "This contact has members of a company group with"
+                    f" different pricelists, the members are:\n{members_str}"
+                ),
```
Comment (thienvh332) [inline sale_partner_company_group/models/res_partner.py]:
```suggestion
                "title": self.env._("Warning"),
                "message": self.env._(
                    "This contact has members of a company group with"
                    f" different pricelists, the members are:\n{members_str}"
                ),
```
```suggestion
                "title": _("Warning"),
                "message": _(
                    "This contact has members of a company group with"
                    f" different pricelists, the members are:\n{members_str}"
                ),
```

File: sale_partner_company_group/models/res_partner.py [hunk 7]
```diff
@@ -0,0 +1,60 @@
+# Copyright 2019 Camptocamp SA
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
+
+from odoo import _, api, models
```
Comment (thienvh332) [inline sale_partner_company_group/models/res_partner.py]:
```suggestion
from odoo import api, models
```

**Review by thienvh332 — COMMENTED** [review]
Some suggested changes, you can ignore it

**Comment by thienvh332** [comment]
Hi @nguyenminhchien 
- We are focusing on porting things that are in **`14.0`** but not in **`17.0`** and then migrating to **`18.0`**. The purpose is to support the project migration from **`v14`** -> **`v18`**. 
- But porting missing PRs from version **`15.0`** is welcome. 
- According to the flow, we will have 2 PRs:
   - **`1 port PR`**
   - **`1 migration PR.`**
  
- **`FYI`**, Currently the modules you can create PR ports include:
  - **`base_partner_company_group + sale_partner_company_group`** (changed: **`PR1587`**)

**Comment by nguyenminhchien** [comment]
> Hi @nguyenminhchien
> 
>     * We are focusing on porting things that are in **`14.0`** but not in **`17.0`** and then migrating to **`18.0`**. The purpose is to support the project migration from **`v14`** -> **`v18`**.
> 
>     * But porting missing PRs from version **`15.0`** is welcome.
> 
>     * According to the flow, we will have 2 PRs:
>       
>       * **`1 port PR`**
>       * **`1 migration PR.`**
> 
>     * **`FYI`**, Currently the modules you can create PR ports include:
>       
>       * **`base_partner_company_group + sale_partner_company_group`** (changed: **`PR1587`**)

It's not easy in this case, the porting commits causes conflicts. So, cannot create 2 separated PRs

File: account_partner_company_group/readme/CREDITS.md (hunk 1 above)
Comment (nguyenminhchien) [reply account_partner_company_group/readme/CREDITS.md:1]: My contribution is not much enough for adding my name to the contributors. so, let it as it be.

**Comment by thienvh332** [comment]
> It's not easy in this case, the porting commits causes conflicts. So, cannot create 2 separated PRs

In this case:
 - We can intervene to split **`PR 1587`** into 2 parts and create a PR for it. 
 - Or we can also create a PR to change both at the same time.
 
Of course, if you feel it takes too much time, we can skip it. Sometimes we may deviate from the process. Making life easier. :smile:

**Review by thienvh332 — APPROVED** [review]
LGTM

**Pull request closed without merging**